File: codewars/7kyu/tidy-number-(special-number-seires-9)/dduck-tidy-number-(special-number-seires-9).py
# https://www.codewars.com/kata/5a87449ab1710171300000fd/train/javascript

from unittest import TestCase


def tidyNumber(n):
    digits = str(n)
    return len(digits) - 1 == sum([True for i in range(0, len(digits) - 1) if int(digits[i]) <= int(digits[i + 1])])

#
def extra_perfect(n):
    return list(range(1,n+1,2))
#

TestCase().assertEqual(tidyNumber(12), True)
TestCase().assertEqual(tidyNumber(102), False)
TestCase().assertEqual(tidyNumber(9672), False)
TestCase().assertEqual(tidyNumber(2789), True)

File: codewars/7kyu/tidy-number-(special-number-seires-9)/dduck-tidy-number-(special-number-seires-9).js
// https://www.codewars.com/kata/5a87449ab1710171300000fd/train/javascript

const tidyNumber = (n) => String(n).split('')
  .every((e, i, arr) => {
    return arr[i + 1] ? e <= arr[i + 1] : true;
  });

console.log(tidyNumber(12), true);
console.log(tidyNumber(102), false);
console.log(tidyNumber(9672), false);
console.log(tidyNumber(2789), true);
console.log(tidyNumber(2335), true);
